### ISSUE TITLE
How to use cmdline.d on host without creating cmdline.d/90crypt.conf on initramfs

### ISSUE THREAD

**Issue by Neurognostic** [dracutdevs/dracut#727]
## Issue

How can I use the `/etc/cmdline.d/` directory on the host for specifying kernel parameters without `/etc/cmdline.d/90crypt.conf` being created on the initramfs?

### Details

When my root device is a LUKS encrypted volume and `hostonly=yes` is set the `crypt` module automatically creates `/etc/cmdline.d/90crypt.conf` on the initramfs with the line `rd.luks.uuid=<MY-UUID>`. This causes an error to printed during boot right on top of where I type my LUKS passphrase when `dracut-initqueue` is trying to find root, because root was already found by `systemd-cryptsetup-generator(8)` from kernel parameter specified elsewhere (e.g. boot loader configuration or with the `kernel_cmdline` option in `dracut.conf(5)`).

```shell
dracut-initqueue[518]: Failed to start systemd-cryptsetup@luksx2d<<CRYPT-UUID>>.service: Unit systemd-cryptsetup@luksx2d<<CRYPT-UUID>>.service: Unit systemd-cryptsetup@luksx2d<<CRYPT-UUID>>.service not found.
```

The error does not prevent my root LUKS volume from being opened and mounted.

---

The only way I have found to prevent this error is to either use `hostonly=no` *or* `hostonly_cmdline=no`, but when I use either of these options `dracut(8)` no longer parses the `cmdline.d` directory. Which forces me to use the `dracut.conf(5)` option `kernel_cmdline` when embedding kernel parameters.

Because of this, the `cmdline.d` directory is useless. I have not found anyway to prevent `cmdline.d/90crypt.conf` from being created on the initramfs while still allowing `cmdline.d/` to be parsed with `hostonly_cmdline=yes`.

This issue is occurs with the latest master branch.

**Comment by haraldh** [comment]
might be fixed with 16abea2592d7fee6166660098b647356d5a4cc14

Btw, systemd does not read `/etc/cmdline.d`.

**Comment by diabonas** [comment]
See your previous issue #727, I think the
```
dracut-initqueue[518]: Failed to start systemd-cryptsetup@luksx2d<<CRYPT-UUID>>.service: Unit systemd-cryptsetup@luksx2d<<CRYPT-UUID>>.service: Unit systemd-cryptsetup@luksx2d<<CRYPT-UUID>>.service not found.
```
error should hopefully be fixed by #794.
